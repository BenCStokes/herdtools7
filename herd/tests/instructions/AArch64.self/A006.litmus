AArch64 A006
"PodWW Irfe PodRR Ifre PodWW Irfe PodRR Ifre"
Generator=diyone7 (version 7.56+02~dev)
Com=Irf Ifr Irf Ifr
Orig=PodWW Irfe PodRR Ifre PodWW Irfe PodRR Ifre
{
0:X0=NOP; 0:X1=P3:L05; 0:X2=P1:Lself01;
2:X0=NOP; 2:X1=P1:L01; 2:X2=P3:Lself03;
}
 P0          | P1        | P2          | P3        ;
 STR W0,[X1] | Lself01:  | STR W0,[X1] | Lself03:  ;
 STR W0,[X2] | B L00     | STR W0,[X2] | B L04     ;
             | MOV W0,#2 |             | MOV W0,#2 ;
             | B L01     |             | B L05     ;
             | L00:      |             | L04:      ;
             | MOV W0,#1 |             | MOV W0,#1 ;
             | L01:      |             | L05:      ;
             | B L02     |             | B L06     ;
             | MOV W1,#2 |             | MOV W1,#2 ;
             | B L03     |             | B L07     ;
             | L02:      |             | L06:      ;
             | MOV W1,#1 |             | MOV W1,#1 ;
             | L03:      |             | L07:      ;
exists (1:X0=2 /\ 1:X1=1 /\ 3:X0=2 /\ 3:X1=1)
