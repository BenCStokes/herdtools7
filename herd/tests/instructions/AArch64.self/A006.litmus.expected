Test A006 Allowed
States 16
1:X0=1; 1:X1=1; 3:X0=1; 3:X1=1;
1:X0=1; 1:X1=1; 3:X0=1; 3:X1=2;
1:X0=1; 1:X1=1; 3:X0=2; 3:X1=1;
1:X0=1; 1:X1=1; 3:X0=2; 3:X1=2;
1:X0=1; 1:X1=2; 3:X0=1; 3:X1=1;
1:X0=1; 1:X1=2; 3:X0=1; 3:X1=2;
1:X0=1; 1:X1=2; 3:X0=2; 3:X1=1;
1:X0=1; 1:X1=2; 3:X0=2; 3:X1=2;
1:X0=2; 1:X1=1; 3:X0=1; 3:X1=1;
1:X0=2; 1:X1=1; 3:X0=1; 3:X1=2;
1:X0=2; 1:X1=1; 3:X0=2; 3:X1=1;
1:X0=2; 1:X1=1; 3:X0=2; 3:X1=2;
1:X0=2; 1:X1=2; 3:X0=1; 3:X1=1;
1:X0=2; 1:X1=2; 3:X0=1; 3:X1=2;
1:X0=2; 1:X1=2; 3:X0=2; 3:X1=1;
1:X0=2; 1:X1=2; 3:X0=2; 3:X1=2;
Ok
Witnesses
Positive: 1 Negative: 15
Condition exists (1:X0=2 /\ 1:X1=1 /\ 3:X0=2 /\ 3:X1=1)
Observation A006 Sometimes 1 15
Hash=edec135b7ea827ac20b4c3a788c485de

